h2/src/main/org/h2/mvstore/db/MVTable.java

```diff
@@ -131,6 +131,7 @@ public boolean lock(Session session, boolean exclusive,
             boolean forceLockEvenInMvcc) {
         int lockMode = database.getLockMode();
         if (lockMode == Constants.LOCK_MODE_OFF) {
+            session.registerTable(this);
```

Comment:
what am I missing here? if lock mode is off, we add the table to Session#locks, but in Session#getLocks(), if lock mode is off, we ignore the contents of the Session#locks ArrayList ?

Reply:
Yes, tables are collected anyway to be able mark them as updated, but for reporting purposes (getLocks()), since there are no locks in off mode, we do not disclose them.